src/Maestro/DependencyUpdateErrorProcessor/DependencyUpdateErrorProcessor.cs

```diff
@@ -0,0 +1,485 @@
+                        $@"Updating a gitHub issue number : '{issueNumber}' for the error : '{updateHistoryError.ErrorMessage}'
+                         for the repository : '{updateHistoryError.Repository}'");
+                    await UpdateIssueAsync(client, updateHistoryError, shouldReplaceDescription,
+                        description, issue, repo.Id);
+                    return;
+                }
+            }
+            // Create a new issue for the error if the issue is already closed or the issue does not exists.
+                _logger.LogInformation(
+                    $@"Creating a new gitHub issue for dependency Update Error, 
+                    for the error message  '{updateHistoryError.ErrorMessage}'
+                    for the repository : '{updateHistoryError.Repository}'");
+                await CreateIssueAsync(client, updateHistoryError, gitHubIssueEvaluator, description, repo.Id,
+                    issueRepo);
+        }
+
+        /// <summary>
+        ///     Parse out the owner and repo from a repository url
+        /// </summary>
+        /// <param name="uri">Github repository URL</param>
+        /// <returns>Tuple of owner and repo</returns>
+        public static (string owner, string repo) ParseRepoUri(string uri)
+        {
+            var u = new UriBuilder(uri);
+            Match match = RepositoryUriPattern.Match(u.Path);
+            if (!match.Success)
+            {
+                return default;
+            }
+            return (match.Groups["owner"].Value, match.Groups["repo"].Value);
+        }
+
+        /// <summary>
+        /// Creates a new github issue.
+        /// </summary>
+        /// <param name="client">Github client</param>
+        /// <param name="updateHistoryError">Error info for which github issue has to be created</param>
+        /// <param name="gitHubIssueEvaluator">Reliable service that holds the (key - repo-branch) and (value issueNumber)</param>
+        /// <param name="description">Description for the issue body or comment body</param>
+        /// <param name="repoId">Repository Id</param>
+        /// <param name="issueRepo">Repository where issue has to be created.</param>
+        /// <returns></returns>
+        private async Task CreateIssueAsync(GitHubClient client, RepositoryBranchUpdateHistoryEntry updateHistoryError,
+            IReliableDictionary<(string repository, string branch),int> gitHubIssueEvaluator, string description, long repoId, string issueRepo)
+        {
+            string labelName = "DependencyUpdateError";
+            NewIssue newIssue = new NewIssue("[Dependency Update] Errors during dependency updates to :" +
+                updateHistoryError.Repository);
+            string bodyTitle = $@"The following errors have been detected when attempting to update dependencies in 
+                '{updateHistoryError.Repository}' 
+                 {Environment.NewLine}";
+            if (!string.IsNullOrEmpty(description))
```

Comment:
Wait... if the description is empty we don't actually create the issue?  That's... unexpected behavior.  Why are the first three lines outside of this check? In fact, why are we calling this method at all if we aren't going to do anything?

Reply:
Description can be empty for the following cases

1)SynchronizePullRequest method - we do not want to capture this as this is not an error 
2) UpdateAssetsAsync - when the subscriptionId is no longer valid, meaning it is deleted.

For all the other scenarios for eg : UpdateAssestsAsync, ProcessPendingUpdatesAsync and default methods the descriptions gets populated. We call this method to get the description for the issue body.  So I have to do this check so that I do not create issues for them. Also I have logging in those methods that we are skipping this because the subscription is invalid or if it is syncPullrequest method.